crypto/src/proofs/transparent.rs

```diff
@@ -439,23 +443,18 @@ impl SwapProof {
             return Err(Error::TransmissionKeyMismatch);
         }
 
+        // TODO: do we need to check the swap NFT Asset ID is properly constructed here?
+
         // Merkle path integrity.
         self.note_commitment_block_proof
             .verify(anchor)
             .map_err(|_| Error::MerkleRootMismatch)?;
 
         // Value commitment integrity.
-        if self.value.commit(self.v_blinding) != value_commitment {
+        if self.swap_nft_value.commit(self.v_blinding) != swap_nft_value_commitment {
```

Comment:
The value commitment integrity check we want to do is different from this: we're going to be consuming the value recording the swap NFT and producing new output notes entirely _internally_ to the `SwapClaim` action, so the only effect on the transaction's value balance is the fee.

We want to check that the `value_commitment` is a commitment to `f` `penumbra` tokens, where `f` is the fee amount committed to by the swap nft's asset id.